core/src/main/java/org/apache/iceberg/rest/requests/CommitTableRequest.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.rest.requests;
+
+import java.util.List;
+import javax.annotation.Nullable;
+import org.apache.iceberg.MetadataUpdate;
+import org.apache.iceberg.UpdateRequirement;
+import org.apache.iceberg.catalog.TableIdentifier;
+import org.apache.iceberg.rest.RESTRequest;
+import org.immutables.value.Value;
+
+@Value.Immutable
+public interface CommitTableRequest extends RESTRequest {
```

Comment:
I'm wondering if we need a new `CommitTableRequest` or if it would be fine to continue using `UpdateTableRequest`. The breaking change is going to be removing the `List<UpdateTableRequest.UpdateRequirement> requirements()` method, but the `UpdateRequirement` interface is also deprecated and will be removed.

Since source compatibility is already going to break, I don't see a problem with moving directly from `List<UpdateTableRequest.UpdateRequirement> requirements()` to `List<UpdateRequirement> requirements()`. The requirements that are returned will implement both interfaces until the older `UpdateTableRequest.UpdateRequirement` is removed.

Reply:
if we 're ok breaking source compability, then we can update `UpdateTableRequest` directly and don't need `CommitTableRequest`